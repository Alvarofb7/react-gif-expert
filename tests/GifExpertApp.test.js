import { fireEvent, render, screen } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";

describe('Pruebas de <GifExpertApp />', () => {

    test('Debe de aparecer la nueva categoria en pantalla si se aÃ±ade', () => {
        const { container} = render(<GifExpertApp />);
        const input = screen.getByRole("textbox");
        const form = screen.getByRole("form");
        fireEvent.input(input, { target: { value: "Categoria" } });
        fireEvent.submit(form);

        expect(screen.getByText("Categoria")).toBeTruthy();
        expect(container).toMatchSnapshot();
    });

});