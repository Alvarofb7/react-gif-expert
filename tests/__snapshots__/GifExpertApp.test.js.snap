// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifExpertApp /> Debe de aparecer la nueva categoria en pantalla si se a√±ade 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    Categoria
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
  <h3 />
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
